libkmod/libkmod-module.c

```diff
@@ -670,21 +671,13 @@ static int do_init_module(struct kmod_module *mod, unsigned int flags, const cha
 			return err;
 		}
 
-		if (flags & KMOD_INSERT_FORCE_MODVERSION) {
-			err = kmod_elf_strip_section(elf, "__versions");
-			if (err < 0)
-				INFO(mod->ctx, "Failed to strip modversion: %s\n",
-				     strerror(-err));
-		}
-
-		if (flags & KMOD_INSERT_FORCE_VERMAGIC) {
-			err = kmod_elf_strip_vermagic(elf);
-			if (err < 0)
-				INFO(mod->ctx, "Failed to strip vermagic: %s\n",
-				     strerror(-err));
-		}
-
-		mem = kmod_elf_get_memory(elf);
+		stripped = kmod_elf_strip(elf, flags);
+		if (stripped == NULL) {
+			INFO(mod->ctx, "Failed to strip version information: %s\n",
+			     strerror(errno));
+			mem = kmod_elf_get_memory(elf);
```

Comment:
separate fix, but we are missing an error return here:  turn that INFO into error and return instead of continuing.

Reply:
And removed. Some tests fail and I can't reproduce it on my system.

Reply:
See https://github.com/kmod-project/kmod/pull/176 for details